### PR TITLE
Milestone 6 - Create a Database Based on a Diagram

### PR DESCRIPTION
**Pull request by mukhammadrizooff**
# In this Pull Request we Implement:
- We take a look at the following database diagram for a clinic:
     -   [Link to see diagram](https://github.com/david-lafontant/vet-clinic/tree/diagram-db)
- We create a file named `schema_based_on_diagram.sql` where we implement the database from the diagram.
- Join tables from many-to-many relationships might not appear in the diagram.
- We add the FK indexes.

### CODE REVIEW

**Review by Lordkaito — CHANGES_REQUESTED** [review]
Hi @mukhammadrizooff and @david-lafontant, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a link that does not include any of the requested features. Please take a look at the link you submitted and ask for a new review,

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by david-lafontant** [comment]
@Lordkaito thank you for the time taken with our project. Please could you be more specific because we don't fully understand. Is the link shared (the PR)? the project requirements? or else?

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "name" VARCHAR(255) NOT NULL,
+    "date_of_birth" DATE NOT NULL,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE "medical_histories"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "admitted_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "patient_id" INTEGER NOT NULL,
```
Comment (keroloussamy) [inline schema_based_on_diagram.sql:13]: Kindly you have to add INDEX for this FOREIGN KEY, as the requirements say.

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "status" VARCHAR(255) NOT NULL,
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE "invoices"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "total_amount" DECIMAL(8, 2) NOT NULL,
+    "generated_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "payed_at" TIMESTAMP(0) WITHOUT TIME ZONE NOT NULL,
+    "medical_history_id" INTEGER NOT NULL,
```
Comment (keroloussamy) [inline schema_based_on_diagram.sql:23]: Kindly you have to add INDEX for this FOREIGN KEY, as the requirements say.

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE "innvoice_items"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "unit_price" DECIMAL(8, 2) NOT NULL,
+    "quantity" INTEGER NOT NULL,
+    "total_price" DECIMAL(8, 2) NOT NULL,
+    "invoice_id" INTEGER NOT NULL,
+    "treatment_id" INTEGER NOT NULL,
```
Comment (keroloussamy) [inline schema_based_on_diagram.sql:33]: Kindly you have to add INDEX for these two FOREIGN KEYs, as the requirements say.

File: schema_based_on_diagram.sql [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    PRIMARY KEY(id)
+);
+
+CREATE TABLE "treatments"(
+    "id" INT GENERATED ALWAYS AS IDENTITY,
+    "type" VARCHAR(255) NOT NULL,
+    "name" VARCHAR(255) NOT NULL,
+    PRIMARY KEY(id)
+);
+
```
Comment (keroloussamy) [inline schema_based_on_diagram.sql:43]: - [ ] Kindly there's a many-to-many relationship between `medical_histories` and `treatments` should be created. You can create `medical_histories_has_treatments` table and add to it the Foreign keys.
- [ ] After that you need to add these two Foreign keys indexes, to speed the performance.

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (david-lafontant) [reply schema_based_on_diagram.sql:13]: Asked change implemented

File: schema_based_on_diagram.sql (hunk 2 above)
Comment (david-lafontant) [reply schema_based_on_diagram.sql:23]: Asked change implemented

File: schema_based_on_diagram.sql (hunk 3 above)
Comment (david-lafontant) [reply schema_based_on_diagram.sql:33]: Asked change implemented

File: schema_based_on_diagram.sql (hunk 4 above)
Comment (david-lafontant) [reply schema_based_on_diagram.sql:43]: Asked change implemented

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello @mukhammadrizooff  and @david-lafontant , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**